### PR TITLE
[WIP]fix detection models pytorch inference

### PR DESCRIPTION
**Pull request by maslovaz**

### CODE REVIEW

**Comment by a-sidorova** [comment]
Я в свом PR'e не добавляла поддержку иных задач кроме классификации. Это был первый этап поддержки PyTorch.
Хотелось бы просто, чтобы вывод инференса можно было обработать общим подходом из `IOAdapter`. Для этого нужно специально подготовить данные для адаптера. Пока только классификация готова - `Softmax` на выходе. Такой же сценарий и для `MXNet`.
Одна из следующих задач - поддержать и другие задачи. Думаю, детекция не решается просто удалением вычисления `Softmax`, поскольку тут должна сломаться классификация :)

**Comment by maslovaz** [comment]
@a-sidorova я не трогаю режим валидации, только инференс, и так он для всех задач работает, которые на картинках запускаются, в частности detection

**Comment by valentina-kustikova** [comment]
> @a-sidorova я не трогаю режим валидации, только инференс, и так он для всех задач работает, которые на картинках запускаются, в частности detection

@maslovaz, есть два момента, если вносить такое изменение:

1. Мы выбрасываем преобразование softmax из замера времени при решении задачи классификации, что вообще-то неправильно, поскольку в остальных сетях оно выполняется + мы получаем на выходе заведомо невалидный результат классификации, как писала @a-sidorova.
1. Никто не гарантирует, что модели для детектирования объектов содержат все финальные слои (у TensorFlow мы такие модели видели). Если мы измеряем только проход по сети, выбрасывая эти слои, то полученный результат вообще не имеет смысла сравнивать с другими фреймворками.

Поэтому я поддерживаю @a-sidorova, что надо смотреть, какой выход у сети, а просто выбрасывать преобразования нельзя.

**Comment by valentina-kustikova** [comment]
@maslovaz, дело не в сохранении результатов инференса, а в отбрасывании преобразования из замера времени инференса, - это принципиально разные  вещи.

@FenixFly

**Comment by ivikhrev** [comment]
> Поэтому я поддерживаю @a-sidorova, что надо смотреть, какой выход у сети, а просто выбрасывать преобразования нельзя.

Если стоит цель измерить производительность фреймворка, то некорректно в замеры включать какие-то преобразования которые выполняются third-party библиотеками.

**Comment by valentina-kustikova** [comment]
У нас цель не просто померить производительность фреймворка, а сравнивать производительность фреймворков.

**Comment by valentina-kustikova** [comment]
Да, речь идёт про одну модель, для которой в некоторых фреймворках последние преобразования могут быть включены в сеть, а в каких-то фреймворках - не включены, поскольку это обычные функции, которые могут быть вызваны для выхода.

**Comment by ivikhrev** [comment]
> последние преобразования могут быть включены в сеть, а в каких-то фреймворках - не включены,

Можно же ориентироваться на исходный фреймворк, из которого пришла модель. Если там этих операций нет внутри самой сети, то и не надо их заносить при конвертации (не знаю, есть ли случаи, где этого нельзя избежать) ? А сам бенчмарк не должен выполнять никаких постпроцессинговых операций.
